src/MongoDB.Driver.Core/Core/Servers/HeartbeatDelay.cs

```diff
@@ -30,7 +30,6 @@ internal sealed class HeartbeatDelay : IDisposable
         // constructors
         public HeartbeatDelay(TimeSpan heartbeatInterval, TimeSpan minHeartbeatInterval)
         {
-            if (minHeartbeatInterval > heartbeatInterval) { minHeartbeatInterval = heartbeatInterval; }
```

Comment:
Removing this line seems like a mistake.

It doesn't make sense for the minHeartbeatInterval to be greater than the actual heartbeatInterval...

Or stated in the opposite direction: it doesn't make sense for the actual heartbeatInterval to be less than the minHeartbeatInterval